src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/SystemResourceKey.cs

```diff
@@ -387,16 +383,9 @@ internal static short GetBamlIdBasedOnSystemResourceKeyId(Type targetType, strin
                 srkField = memberName;
             }
 
-            try
-            {
-                srkId = (SystemResourceKeyID)Enum.Parse(typeof(SystemResourceKeyID), srkField);
-            }
-            catch (ArgumentException)
-            {
-                found = false;
-            }
-
-            if (found)
+            if (targetType.Assembly == XamlTypeMapper.AssemblyPF &&
+                targetType.FullName == "System.Windows.SystemParameters" &&
+                Enum.TryParse(srkField, out SystemResourceKeyID srkId))
```

Comment:
Enum.TryParse is not available on .NET 3.0 and 3.5 which support WPF and out variables require C# 7.0. Might be OK, just pointing out this is changing runtime and compilation requirements.

Reply:
This repository is for .NET Core 3+ only, so you can assume thats available.

Reply:
It's SDK code, not runtime. And then, as far as I'm concerned, .Net Framework 3.0 and 3.5 are both out of support at present, so they're impossible to get this change, especially runtime. What's more, it's unnecessary to build WPF app with SDK source code referenced. So it doesn't matter to use C# 7.0+. :)